File: algosbradfield/queue.js
var Queue = function() {
    this.items = []
}

Queue.prototype.is_empty = function() {
    return this.items.length === 0
}

Queue.prototype.enqueue = function(item) {
    this.items.unshift(item)
}

Queue.prototype.dequeue = function() {
    return this.items.pop()
}

Queue.prototype.size = function() {
    return this.items.length
}

var q = new Queue()
q.enqueue('foo')
q.enqueue('bar')
console.log(q.size())
console.log(q.dequeue())
console.log(q.dequeue())
console.log(q.dequeue())

File: algosbradfield/queue.py
class Queue():

    def __init__(self):
        self._items = []

    def is_empty(self):
        return self._items == []

    def enqueue(self, item):
        self._items.insert(0, item)

    def dequeue(self):
        try:
            return self._items.pop()
        except IndexError:
            return 'empty!'
    
    def size(self):
        return len(self._items)


# But it's better to use 
# collections.deque for O(1) enqueues and dequees

q = Queue()
q.enqueue('foo')
q.enqueue('bar')
print(q.dequeue())
print(q.dequeue())
print(q.dequeue())